tests/integration_tests/modules/test_apt.py

```diff
@@ -0,0 +1,291 @@
+"""Series of integration tests covering apt functionality."""
+import re
+from tests.integration_tests.clouds import ImageSpecification
+
+import pytest
+
+from tests.integration_tests.instances import IntegrationInstance
+
+
+USER_DATA = """\
+#cloud-config
+apt:
+  conf: |
+    APT {
+        Get {
+            Assume-Yes "true";
+            Fix-Broken "true";
+        }
+    }
+  proxy: "http://proxy.internal:3128"
+  http_proxy: "http://squid.internal:3128"
+  ftp_proxy: "ftp://squid.internal:3128"
+  https_proxy: "https://squid.internal:3128"
+  primary:
+    - arches: [default]
+      uri: http://badarchive.ubuntu.com/ubuntu
+  security:
+    - arches: [default]
+      uri: http://badsecurity.ubuntu.com/ubuntu
+  sources_list: |
+    deb $MIRROR $RELEASE main restricted
+    deb-src $MIRROR $RELEASE main restricted
+    deb $PRIMARY $RELEASE universe restricted
+    deb-src $PRIMARY $RELEASE universe restricted
+    deb $SECURITY $RELEASE-security multiverse
+    deb-src $SECURITY $RELEASE-security multiverse
+  sources:
+    test_keyserver:
+        keyid: 72600DB15B8E4C8B1964B868038ACC97C660A937
+        keyserver: keyserver.ubuntu.com
+        source: "deb http://ppa.launchpad.net/cloud-init-raharper/curtin-dev/ubuntu $RELEASE main"
+    test_ppa:
+      keyid: 3552C902B4DDF7BD38421821015D28D7441614D8
```

Comment:
One thing I noticed is that in the new tests we are using the full fingerprint not a short 8 character fingerprint. It's not critical, but since we have two keyid examples, we could have one use a short fingerprint instead so we are exercising more failure paths (even though it's not really a cloud-init failure path but a tooling path)